### PR TITLE
fix: using href="#*" spams browser history

### LINKED ISSUE #389: Using href="#***" in tab spams browser history
#### Composer packages

```
$ composer show
# Put the result here.
```

#### PHP version

```
PHP 8.1.27 (cli) (built: Mar 12 2024 05:04:18) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.27, Copyright (c) Zend Technologies
    with Xdebug v3.2.2, Copyright (c) 2002-2023, by Derick Rethans
```

## Subject

Using href="#***" in translation tabs spams browser history with no reason.
When we switch language tab it calls `history.pushState(**)` with tab `#anchor`, but actually we dont need that, as page does not re-rendering to actual tab.

If package used with easyadmin, to get back to the index page after switching language tab, we have to press `*Back browser button* <-` multiple times.

## Steps to reproduce
Open crud page, switch language tab multiple times, press "Back" button in your browser

## Expected results
Go back to the crud index page

## Actual results

Url changed, but nothing on page changed actually

### PR DESCRIPTION
**Pull request by karrakoliko**
I am targeting this branch, because i don't see these files in 2.x.

Closes #389

## Changelog

```markdown

### Added

### Changed
Replaced `<a href="*">` to `<div data-bs-target="*">`

### Deprecated

### Removed

### Fixed

### Security
```

## To do

- [ ] Update the tests
- [ ] Update the documentation
- [ ] Add an upgrade note

### CODE REVIEW

File: src/Resources/views/bootstrap_4_layout.html.twig [hunk 1]
```diff
@@ -7,11 +7,11 @@
             {% set locale = translationsFields.vars.name %}
 
             <li class="nav-item">
-                <a href="#{{ translationsFields.vars.id }}_a2lix_translations-fields" class="nav-link {% if app.request.locale == locale %}active{% endif %}" data-toggle="tab" role="tab">
+                <div data-bs-target="#{{ translationsFields.vars.id }}_a2lix_translations-fields" class="nav-link {% if app.request.locale == locale %}active{% endif %}" data-toggle="tab" role="tab">
```
Comment (webda2l) [inline src/Resources/views/bootstrap_4_layout.html.twig:10]: [data-target](https://getbootstrap.com/docs/4.6/components/navs/#using-data-attributes) for v4

File: src/Resources/views/macros.html.twig [hunk 2]
```diff
@@ -16,11 +16,11 @@ Example:
             {% set locale = translationsFields.vars.name %}
 
             <li class="nav-item">
-                <a href="#{{ translationsFields.vars.id }}_{{ fieldsNames|join('_') }}_a2lix_translations-fields" class="nav-link {% if app.request.locale == locale %}active{% endif %}" data-toggle="tab" role="tab">
+                <div data-bs-target="#{{ translationsFields.vars.id }}_{{ fieldsNames|join('_') }}_a2lix_translations-fields" class="nav-link {% if app.request.locale == locale %}active{% endif %}" data-toggle="tab" role="tab">
```
Comment (webda2l) [inline src/Resources/views/macros.html.twig:19]: Ignore this file

File: src/Resources/views/bootstrap_3_layout.html.twig [hunk 3]
```diff
@@ -7,11 +7,11 @@
                 {% set locale = translationsFields.vars.name %}
 
                 <li{% if app.request.locale == locale %} class="active"{% endif %}>
-                    <a href="#{{ translationsFields.vars.id }}_a2lix_translations-fields" data-toggle="tab" role="tab">
+                    <div data-bs-target="#{{ translationsFields.vars.id }}_a2lix_translations-fields" data-toggle="tab" role="tab">
```
Comment (webda2l) [inline src/Resources/views/bootstrap_3_layout.html.twig:10]: Ignore this file, v3 is not maintained anymore and this file should drop

File: src/Resources/views/macros_bootstrap_3.html.twig [hunk 4]
```diff
@@ -14,11 +14,11 @@ Example:
                 {% set locale = translationsFields.vars.name %}
 
                 <li{% if app.request.locale == locale %} class="active"{% endif %}>
-                    <a href="#{{ translationsFields.vars.id }}_{{ fieldsNames|join('_') }}_a2lix_translations-fields" data-toggle="tab" role="tab">
+                    <div data-bs-target="#{{ translationsFields.vars.id }}_{{ fieldsNames|join('_') }}_a2lix_translations-fields" data-toggle="tab" role="tab">
```
Comment (webda2l) [inline src/Resources/views/macros_bootstrap_3.html.twig:17]: Ignore this file, v3 is not maintained anymore and this file should drop

File: src/Resources/views/macros_bootstrap_4.html.twig [hunk 5]
```diff
@@ -14,11 +14,11 @@ Example:
                 {% set locale = translationsFields.vars.name %}
 
                 <li class="nav-item">
-                    <a href="#{{ translationsFields.vars.id }}_{{ fieldsNames|join('_') }}_a2lix_translations-fields" class="nav-link {% if app.request.locale == locale %}active{% endif %}" data-toggle="tab" role="tab">
+                    <div data-bs-target="#{{ translationsFields.vars.id }}_{{ fieldsNames|join('_') }}_a2lix_translations-fields" class="nav-link {% if app.request.locale == locale %}active{% endif %}" data-toggle="tab" role="tab">
```
Comment (webda2l) [inline src/Resources/views/macros_bootstrap_4.html.twig:17]: [data-target](https://getbootstrap.com/docs/4.6/components/navs/#using-data-attributes) for v4

**Review by webda2l — CHANGES_REQUESTED** [review]

**Comment by karrakoliko** [comment]
@webda2l done

**Pull request merged**